deps: update sentry dependency

lzap opened the pull request:
This multi-commit PR updates sentry dependency and removes deprecated function call. Then in multiple commits fixes few cosmetic things reported by staticcheck.

Edit: Dropped all the other commits, I will separate them out.

mvo5 reviewed (commented):
The content looks fine, thanks for the cleanup.

I personally think the PR title ("Sentry1") and the individual commit descriptions are a bit short, they are simple enough so maybe fine, but I personally would for example reference "effective go" (with a link) in the commit that changes NoJWTError->ErrNoJWT (and similarly a short rational in the others). Happy to approve if noone else share this sentiment.

lzap commented:
Fixed the PR title. Yeah, no hard feelings about the error prefix, I just fixed all staticcheck complaints. I gave it a little google and I cannot seem to find any mention of that prefix in any of the official docs. But the linter warning was this one:

https://staticcheck.dev/docs/checks#ST1012

supakeen commented:
Approved though I would have preferred this to be two separate PRs (one for sentry, one for the staticcheck fixes) :)

lzap commented:
On my defense, these are actually all linter complaints I just worded it poorly.

lzap commented on cmd/osbuild-composer/main.go:
```diff
@@ -108,7 +108,7 @@ func main() {
 		}
 
-		sentryhook := sentrylogrus.NewFromClient([]logrus.Level{logrus.PanicLevel,
+		sentryhook := sentrylogrus.NewEventHookFromClient([]logrus.Level{logrus.PanicLevel,
```
This is needed, sentry API is not stable yet and the function was renamed in the newer version.

thozza reviewed (approved):
Thanks, LGTM!
